Stripe/NSDictionary+Stripe.h

```diff
@@ -16,6 +16,8 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (NSDictionary *)stp_dictionaryByRemovingNulls;
 
+- (NSDictionary<NSString *, NSString *> *)stp_dictionaryByRemovingNonStrings;
```

Comment:
I don't think we should do this. We don't currently enforce this type for other classes like STPSource and I don't think we should change that or add the restriction here. It is not clear to me from the API docs that they must in fact be string pairs (as opposed to having number, dictionary, or array values). I think we should just leave as `NSDictionary *` or at most `NSDictionary<NSString *, id> *`

Reply:
Hmm, I'm actually for this. It is an API constraint, and though the docs aren't super clear we do say:

> Note: You can specify up to 20 keys, with key names up to 40 characters long and values up to 500 characters long.

[0] https://stripe.com/docs/api#metadata

Reply:
Oh interesting. I didn't realize we specified that. In that case I'm ok with changing our API to match this constraint 👍